### PR TITLE
Added Tree sitter provider

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
## Checklist

- [/] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [/] I have not broken the cheatsheet

### CODE REVIEW

File: packages/cursorless-engine/src/languages/LanguageDefinition.ts [hunk 1]
```diff
@@ -91,43 +90,42 @@ export class LanguageDefinition {
  * @returns The text of the query file, with all imports inlined
  */
 async function readQueryFileAndImports(
-  fileSystem: FileSystem,
-  languageQueryPath: string,
+  ide: IDE,
+  provider: RawTreeSitterQueryProvider,
+  languageQueryName: string,
```
Comment (AndreasArvidsson) [inline packages/cursorless-engine/src/languages/LanguageDefinition.ts:95]: The reason for switching from path to name is that if we have a non file system implementation of the language provider, path is irrelevant. I think every definition name(scm filename) should just be unique and in which folder they are located doesn't really matter.

File: packages/cursorless-engine/src/languages/LanguageDefinitions.ts [hunk 2]
```diff
@@ -1,31 +1,56 @@
...
 import { LanguageDefinition } from "./LanguageDefinition";
-import { toString } from "lodash-es";
 
 /**
  * Sentinel value to indicate that a language doesn't have
  * a new-style query definition file
  */
 const LANGUAGE_UNDEFINED = Symbol("LANGUAGE_UNDEFINED");
 
+export interface LanguageDefinitions {
```
Comment (pokey) [inline packages/cursorless-engine/src/languages/LanguageDefinitions.ts:23]: I removed the lifecycle methods from this interface (`init` and `dispose`). I don't think interfaces should expose lifecycle methods. Clients of an interface shouldn't be responsible for the lifecycle of any given instance; that is the job of whoever constructs the actual class

**Review by pokey — APPROVED** [review]
looks good! See my various changes and related comment below. Merge if you're happy

File: packages/cursorless-engine/src/languages/LanguageDefinitions.ts (hunk 2 above)
Comment (AndreasArvidsson) [reply packages/cursorless-engine/src/languages/LanguageDefinitions.ts:23]: Very true

**Pull request merged**